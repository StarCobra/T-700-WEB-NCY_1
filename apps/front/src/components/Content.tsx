import React from "react";
import { Box } from "@mui/material";

// eslint-disable-next-line prettier/prettier
export default function Content(props: any) {
  const { title = "", content = "" } = props;

  return (
    <Box className="contentContainer">
      {title !== "" && <h3>{title}</h3>}

      {content !== "" && <p>{content}</p>}
    </Box>
  );
}
