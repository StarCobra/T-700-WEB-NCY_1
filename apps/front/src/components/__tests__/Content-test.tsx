import React from "react";
import { act, create } from "react-test-renderer";
import Content from "../Content";

describe("Components > Content", () => {
  const tests = [
    {
      description: "without props",
      props: {},
    },
    {
      description: "with title",
      props: { title: "Beautiful title" },
    },
    {
      description: "with content",
      props: { content: "Content Lorem Ipsum" },
    },
    {
      description: "with title and content",
      props: { title: "Beautiful title", content: "Content Lorem Ipsum" },
    },
  ];
  tests.forEach((test) => {
    it(`should render the component ${test.description}`, async () => {
      let root;
      await act(async () => {
        root = create(<Content {...test.props} />);
      });
      expect(root).toMatchSnapshot();
    });
  });
});
