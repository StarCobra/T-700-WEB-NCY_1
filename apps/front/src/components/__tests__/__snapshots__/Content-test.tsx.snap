// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > Content should render the component with content 1`] = `
<div
  className="contentContainer MuiBox-root css-0"
>
  <p>
    Content Lorem Ipsum
  </p>
</div>
`;

exports[`Components > Content should render the component with title 1`] = `
<div
  className="contentContainer MuiBox-root css-0"
>
  <h3>
    Beautiful title
  </h3>
</div>
`;

exports[`Components > Content should render the component with title and content 1`] = `
<div
  className="contentContainer MuiBox-root css-0"
>
  <h3>
    Beautiful title
  </h3>
  <p>
    Content Lorem Ipsum
  </p>
</div>
`;

exports[`Components > Content should render the component without props 1`] = `
<div
  className="contentContainer MuiBox-root css-0"
/>
`;
